resourcescaler.go

```diff
@@ -48,147 +47,142 @@ func New(kubeconfigPath string, namespace string) (scaler_types.ResourceScaler,
 	}, nil
 }
 
-func (s *AppResourceScaler) SetScale(resource scaler_types.Resource, scaling int) error {
-
-	// get ingress by resource name
-	ingress, err := s.kubeClientSet.ExtensionsV1beta1().Ingresses(s.namespace).Get(string(resource), meta_v1.GetOptions{})
-	if err != nil {
-		s.logger.WarnWith("Failure during retrieval of ingress", "resource_name", string(resource))
-		return errors.Wrap(err, "Failed getting ingress instance")
+func (s *AppResourceScaler) SetScale(resource scaler_types.Resource, scale int) error {
+	if scale == 0 {
+		return s.scaleServiceToZero(s.namespace, string(resource))
 	}
+	return s.scaleServiceFromZero(s.namespace, string(resource))
+}
 
-	ingress.GetObjectMeta().SetAnnotations(map[string]string{
-		"nginx.ingress.kubernetes.io/configuration-snippet": fmt.Sprintf(
-			`proxy_set_header X-App-Target "%s";`, string(resource)),
+func (s *AppResourceScaler) scaleServiceFromZero(namespace string, serviceName string) error {
+	var jsonPatchMapper []map[string]string
+	s.logger.DebugWith("Scaling from zero", "namespace", namespace, "serviceName", serviceName)
+	path := fmt.Sprintf("/spec/spec/tenants/0/spec/services/%s/state", string(serviceName))
+	jsonPatchMapper = append(jsonPatchMapper, map[string]string{
+		"op":    "add",
```

Comment:
update?

Reply:
Talked offline, there's no update op(eration), there's maybe `replace` but it will fail if the field doesn't exist, `add` will create the field if it doesn't exist, or replace its value if it exists